### PR TITLE
[TLM] Add custom types to docstrings

### PR DESCRIPTION
**Pull request by ulya-tkch**
This change will allow our custom defined types to show up on help.cleanlab.ai [here](https://help.cleanlab.ai/reference/python/trustworthy_language_model/)

### CODE REVIEW

File: cleanlab_studio/studio/trustworthy_language_model.py [hunk 1]
```diff
@@ -437,8 +437,10 @@ def get_trustworthiness_score(
             response (str | Sequence[str]): existing response (or list of responses) associated with the input prompts.
                 These can be from any LLM or human-written responses.
         Returns:
-            float | List[float]: float or list of floats (if multiple prompt-responses were provided) corresponding
-                to the TLM's trustworthiness score.
+            TLMScoreResponse | TLMBatchScoreResponse: **TLMScoreResponse** represents a single TLM response that can be either float, representing the trustworthiness score or a TLMScore object containing both the trustworthiness score and log dictionary keys.
+
+                **TLMBatchScoreResponse** (if multiple prompt-responses were provided) represents a TLM response that can be either a list of floats or a list of TLMScore objects. The list will have the be length as the input list of prompts, response pairs.
+
```
Comment (jwmueller) [inline cleanlab_studio/studio/trustworthy_language_model.py:443]: wai

**Comment by jwmueller** [comment]
@ulya-tkch I unfortunately don't find the new descriptions super clear either.  Perhaps you could first explain everything as if it were just floats or list of floats. And then later in the docstring say:

Actually instead of float, this may be a `TLMScore` object if ... (important to specify when it might not be a float / the simpler return type).

Will leave reviews of this to @jas2600 

But encourage you to write as simply and clearly as you can.

**Comment by ulya-tkch** [comment]
[MAX_PROMPT_LENGTH_TOKENS](https://github.com/cleanlab/cleanlab-studio/blob/1f8c95f9da7320a621618b7064e7fefcc094188b/tests/tlm/test_validation.py#L11) needs to be increased for TLM unit tests to start passing again.

File: cleanlab_studio/studio/trustworthy_language_model.py [hunk 2]
```diff
@@ -4,6 +4,14 @@
 **This module is not meant to be imported and used directly.** Instead, use [`Studio.TLM()`](/reference/python/studio/#method-tlm) to instantiate a [TLM](#class-tlm) object, and then you can use the methods like [`prompt()`](#method-prompt) and [`get_trustworthiness_score()`](#method-get_trustworthiness_score) documented on this page.
 
 The [Trustworthy Language Model tutorial](/tutorials/tlm/) further explains TLM and its use cases.
+
+### Type Aliases
+
+Type aliases returned by the TLM module.
+
+- `TLMScoreResponse = Union[float, TLMScore]`: a single TLM response that can be either float, representing the trustworthiness score or a [TLMScore](#class-tlmscore) object containing both the trustworthiness score and log dictionary keys.
+- `TLMBatchScoreResponse = Union[List[float], List[TLMScore]]`: a TLM response that can be either a list of floats or a list of [TLMScore](#class-tlmscore) objects containing both the trustworthiness score and log dictionary keys. The list will have the be length as the input list of prompts, response pairs.
```
Comment (jas2600) [inline cleanlab_studio/studio/trustworthy_language_model.py:13]:
```suggestion
- `TLMBatchScoreResponse = Union[List[float], List[TLMScore]]`: a TLM response that can be either a list of floats or a list of [TLMScore](#class-tlmscore) objects containing both the trustworthiness score and log dictionary keys. The list will have the same length as the input list of prompts, response pairs.
```

File: cleanlab_studio/studio/trustworthy_language_model.py [hunk 3]
```diff
@@ -4,6 +4,14 @@
...
+- `TLMOptionalBatchScoreResponse = Union[List[Optional[float]], List[Optional[TLMScore]]]`:  a TLM response that can be either a list of floats or None (if the call to the TLM failed) or a list of [TLMScore](#class-tlmscore) objects containing both the trustworthiness score and log dictionary keys or None (if the call to the TLM failed). The list will have the be length as the input list of prompts, response pairs.
```
Comment (jas2600) [inline cleanlab_studio/studio/trustworthy_language_model.py:14]:
```suggestion
- `TLMOptionalBatchScoreResponse = Union[List[Optional[float]], List[Optional[TLMScore]]]`:  a TLM response that can be either a list of floats or None (if the call to the TLM failed) or a list of [TLMScore](#class-tlmscore) objects containing both the trustworthiness score and log dictionary keys or None (if the call to the TLM failed). The list will have the same length as the input list of prompts, response pairs.
```

**Review by jas2600 — APPROVED** [review]
two small suggested changes, otherwise LGTM

**Pull request merged**